x/programs/rust/examples/token/src/lib.rs

```diff
@@ -1,27 +1,89 @@
-use wasmlanche_sdk::{public, store::State, types::Address};
+use wasmlanche_sdk::{program::Program, public, types::Address};
+
+
+/// The internal state keys used by the program.
+#[repr(u8)]
+enum StateKey {
+    TotalSupply,
+    Name,
+    Symbol,
+    Balances(Address),
+}
+
+impl StateKey {
+    pub fn to_vec(self) -> Vec<u8> {
+        match self {
+            StateKey::TotalSupply => vec![self.into()],
+            StateKey::Name => vec![self.into()],
+            StateKey::Symbol => vec![self.into()],
+            StateKey::Balances(a) => {
+                let mut bytes = Vec::with_capacity(1 + 32);
```

Comment:
32 should be a const

Reply:
OR... you could use an iterator and let the compiler handle capacities for you, 😉.
```rust
StateKey::Balances(a) => {
    std::iter::once(self as u8).chain(a.as_bytes().copied()).collect()
}
```

Reply:
cool sounds good